apps/src/flowlab/views/flow/nodes/AiNode.tsx

```diff
@@ -26,6 +36,56 @@ function AiNode({
 
   const [isWorking, setIsWorking] = useState(false);
 
+  const channelId = useAppSelector(state => state.lab.channel?.id);
+
+  const uploadedFiles = useRef<ChatAsset[]>([]);
+  const [uploadedFileCount, setUploadedFileCount] = useState(0);
+
+  const onUploadFiles = async (event: ChangeEvent<HTMLInputElement>) => {
+    console.log('onUploadFiles called');
+    const files = event.target.files;
+    if (!files) {
+      return;
+    }
+
+    const allowedFiles = Array.from(files).map<[string, ChatAsset, File]>(
+      file => [
+        // Create a unique key for each upload in case the same file is uploaded more than once.
+        `${file.name}-${Date.now()}`,
+        {filename: file.name, source: AssetSource.PROJECT},
+        file,
+      ]
+    );
+
+    for (const [, asset, file] of allowedFiles) {
+      try {
+        await HttpClient.put(getAssetUrl(asset, channelId), file);
+
+        uploadedFiles.current.push(asset);
+        setUploadedFileCount(uploadedFiles.current.length);
+      } catch (error) {
+        let status: 'sizeLimitExceeded' | 'uploadFailed' = 'uploadFailed';
+        if (error instanceof NetworkError && error.response.status === 413) {
+          status = 'sizeLimitExceeded';
+        } else {
+          status = 'uploadFailed';
+        }
```

Comment:
nit: can get rid of these `else`, or even convert this logic to a ternary

```
const status = (error instanceof NetworkError && error.response.status === 413) ? 'sizeLimitExceeded' : 'uploadFailed'
```

Reply:
Good call.